src/main/java/mate/academy/controller/CategoryController.java

```diff
@@ -0,0 +1,94 @@
+package mate.academy.controller;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import jakarta.validation.constraints.Positive;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import mate.academy.dto.book.BookDtoWithoutCategoryIds;
+import mate.academy.dto.category.CategoryDto;
+import mate.academy.dto.category.CategoryRequestDto;
+import mate.academy.service.BookService;
+import mate.academy.service.CategoryService;
+import org.springdoc.core.annotations.ParameterObject;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.web.PageableDefault;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "Category management", description = "Endpoints for mapping categories")
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("/categories")
+@Validated
+public class CategoryController {
+    private final BookService bookService;
+    private final CategoryService service;
+
+    @PostMapping
+    @PreAuthorize("hasRole('ADMIN')")
+    @Operation(summary = "Create a new category")
+    public CategoryDto createCategory(@RequestBody @Valid CategoryRequestDto category) {
+        return service.save(category);
+    }
+
+    @PostMapping("/{id}")
+    @PreAuthorize("hasRole('ADMIN')")
+    @Operation(
+            summary = "Update existing category",
+            description = """
+                    Update existing category by its id and specifying new parameters"""
+    )
+    public CategoryDto updateCategory(@PathVariable @Positive Long id,
+                                      @RequestBody @Valid CategoryRequestDto category) {
+        return service.update(id, category);
+    }
+
+    @GetMapping
+    @PreAuthorize("hasAnyRole('ADMIN', 'USER')")
```

Comment:
can this be simplified to `hasRole(USER)`? because I assume that any user has `USER` role

in case of admin - they have both `USER` and `ADMIN` roles

correct me if I'm wrong

Reply:
I have an admin (only admin rights) and users. I thought the admin wouldn't log in through the UI.